src/main/resources/org/jenkinsci/plugins/mesos/MesosComputer/mesos-slave-agent.jnlp.jelly

```diff
@@ -0,0 +1,75 @@
+<!--
+The MIT License
+
+Copyright (c) 2004-2009, Sun Microsystems, Inc., Kohsuke Kawaguchi
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+-->
+
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson">
+  <st:contentType value="application/x-java-jnlp-file" />
+  <j:new var="h" className="hudson.Functions" />
+  <j:set var="rootURL" value="${app.rootUrl}" />
+
+  <!--
+    See http://www.dallaway.com/acad/webstart/ for obtaining the certificate.
+  -->
+
+    <!-- See http://java.sun.com/j2se/1.5.0/docs/guide/javaws/developersguide/syntax.html for the syntax -->
+    <jnlp spec="1.0+"
+        codebase="${rootURL}computer/${h.encode(it.node.nodeName)}/">
+
+      <information>
+        <title>Slave Agent for ${it.displayName}</title>
```

Comment:
This is the default jelly that’ll be used which passes the URL via JNLP response : https://github.com/maselvaraj/jenkins/blob/master/core/src/main/resources/hudson/slaves/SlaveComputer/slave-agent.jnlp.jelly#L71 
 
Please check line #71 (This is the defualt URL configured in Jenkins) and #85 (This is the URL used to fetch JNLP response. In our case it is the url configured in Mesos plugin). I’ve just used the same jelly file, just removed the Jenkins root URL part, so that it talks only to URL from request.